pkg/controller/tas/node_failure_controller_test.go

```diff
@@ -143,6 +167,37 @@ func TestNodeFailureReconciler(t *testing.T) {
 			reconcileRequests: []reconcile.Request{{NamespacedName: types.NamespacedName{Name: nodeName}}},
 			wantFailedNode:    nodeName,
 		},
+		"Node NotReady, pod terminating, marked as unavailable": {
+			featureGates: []featuregate.Feature{features.TASReplaceNodeOnPodTermination},
+			initObjs: []client.Object{
+				newNodeTest(nodeName, corev1.ConditionFalse, fakeClock, time.Duration(0)),
+				baseWorkload.DeepCopy(),
+				terminatingPod,
+			},
+			reconcileRequests: []reconcile.Request{{NamespacedName: types.NamespacedName{Name: nodeName}}},
+			wantFailedNode:    nodeName,
+		},
+		"Node NotReady, pod failed, marked as unavailable": {
+			featureGates: []featuregate.Feature{features.TASReplaceNodeOnPodTermination},
+			initObjs: []client.Object{
+				newNodeTest(nodeName, corev1.ConditionFalse, fakeClock, time.Duration(0)),
+				baseWorkload.DeepCopy(),
+				failedPod,
+			},
+			reconcileRequests: []reconcile.Request{{NamespacedName: types.NamespacedName{Name: nodeName}}},
+			wantFailedNode:    nodeName,
+		},
```

Comment:
make a copy of this test case, but with feature gate disabled, to make sure nothing happens?